src/Peachpie.CodeAnalysis/CodeGen/Graph/BoundExpression.cs

```diff
@@ -2345,7 +2345,12 @@ internal override TypeSymbol Emit(CodeGenerator cg)
                 else
                 {
                     Debug.Assert(Receiver != null);
-                    throw cg.NotImplementedException();
+
+                    // methodRoutineInfo.BindTarget(Receiver)
+                    m.EmitLoadRoutineInfo(cg);
+                    cg.EmitConvert(Receiver, cg.CoreTypes.Object);
```

Comment:
EmitConvert(`Receiver`,  `T`)

Reply:
What is `T`? The method needs an `object` as an argument.

Reply:
ok, casting to the `T` (the actual method declaring class) would cause a more readable exception in case type analysis was wrong